.changelog/unreleased/improvements/2358-tx-memo-field.md

```diff
@@ -0,0 +1,2 @@
+- Introduce a memo field, to allow including arbitrary data (up to 1 KiB) inside
```

Comment:
Are you still enforcing this 1KiB limit somewhere? Or is this no longer a requirement?